docs/Golden-Gate/Bridge-Walkthrough.rst

```diff
@@ -0,0 +1,144 @@
+Bridge Walkthrough
+======================
+In this section, we'll walkthrough a simple Target-to-Host bridge provided with FireSim
+to demonstrate how to integrate your own. This includes the UARTBridge, which
+uses host-MMIO to model a UART device -- this is what you're using when you
+attach to a FireSim simulator's console.
+
+Reading the Bridges section is a prerequisite to reading these sections.
+
+UART Bridge (Host-MMIO)
+-----------------------
+
+Source code for the UART Bridge lives in the following memory locations:
+
+.. code-block:: text
+
+    sim/
+    ├-firesim-lib/src/main/
+    │                 ├-scala/bridges/UARTBridge.scala # Target-Side Bridge and BridgeModule Definitions
+    │                 ├-cc/brides/uart.cc # Bridge Driver source
+    │                 └-cc/brides/uart.h  # Bridge Driver header
+    ├-src/main/cc/firesim/firesim_top.cc  # Driver instantiation in the main simulation driver
+    └-src/main/makefrag/firesim/Makefrag  # Build system modifications to compile Bridge Driver code
+
+UART Bridge Target-Side Chisel
+++++++++++++++++++++++++++++++
+The first order of business when designing a new bridge is to implement its
+target side. In the case of UART we've defined a Chiesl BlackBox extending Bridge.
+We'll instantiate this BlackBox and connect it to UART IO in the
+top-level of our chip.
+
+We first define a class that captures the target-side interface of the Bridge:
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Target-Side Interface
+    :end-before: DOC include end: UART Bridge Target-Side Interface
+
+Here, we define a case class that carries additional metadata to the host-side
```

Comment:
Maybe explain where does this information come from?

Reply:
I'm not sure i follow. The user and the context in which the bridge is instantiated?